### PR TITLE
fix(react): revert react output target version

### LINKED ISSUE #29848: bug: app.component.html:1 ERROR NotFoundError: Failed to execute 'insertBefore' on 'Node': The node before which the new node is to be inserted is not a child of this node.
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/docs/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

v8.x

### Current Behavior

When creating a new app using the CLI, using **angular standalone**, run npm start, nothing is shown in the browser and the console shows this error:

```
core.mjs:7397 ERROR NotFoundError: Failed to execute 'insertBefore' on 'Node': The node before which the new node is to be inserted is not a child of this node.
    at addStyle (index.js:858:34)
    at attachStyles (index.js:887:20)
    at index.js:1705:5
    at Generator.next (<anonymous>)
    at asyncGeneratorStep (asyncToGenerator.js:3:1)
    at _next (asyncToGenerator.js:17:1)
    at asyncToGenerator.js:22:1
    at new ZoneAwarePromise (zone.js:2702:25)
    at asyncToGenerator.js:14:1
    at updateComponent (index.js:1699:20)
```
## Further investigation
1. removing all the html generated code does not show the error.
2. removing only the ion-app shows the error but also show the page

It feels like there is something wrong in the IonApp component

### Expected Behavior

The app is shown in the browser and no error appears in the console

### Steps to Reproduce

1. run `ionic start test-app`
2. run `npm start` from the app root directory
3. open the browser and navigate to `localhost:4200` (nothing appears, only a black screen)
4. open also the console pressing F12 to see the error

### Code Reproduction URL

https://stackblitz.com/~/github.com/scotch83/test-ionic-angular-standalone

### Ionic Info

**ionic info**
[WARN] Error loading @capacitor/ios package.json: Error: Cannot find module '@capacitor/ios/package.json'
       
       Require stack:
       - /Users/scotch-83/.nvm/versions/node/v22.6.0/lib/node_modules/@ionic/cli/lib/project/index.js
       - /Users/scotch-83/.nvm/versions/node/v22.6.0/lib/node_modules/@ionic/cli/lib/index.js
       - /Users/scotch-83/.nvm/versions/node/v22.6.0/lib/node_modules/@ionic/cli/index.js
       - /Users/scotch-83/.nvm/versions/node/v22.6.0/lib/node_modules/@ionic/cli/bin/ionic
[WARN] Error loading @capacitor/android package.json: Error: Cannot find module '@capacitor/android/package.json'
       
       Require stack:
       - /Users/scotch-83/.nvm/versions/node/v22.6.0/lib/node_modules/@ionic/cli/lib/project/index.js
       - /Users/scotch-83/.nvm/versions/node/v22.6.0/lib/node_modules/@ionic/cli/lib/index.js
       - /Users/scotch-83/.nvm/versions/node/v22.6.0/lib/node_modules/@ionic/cli/index.js
       - /Users/scotch-83/.nvm/versions/node/v22.6.0/lib/node_modules/@ionic/cli/bin/ionic

Ionic:

   Ionic CLI                     : 7.2.0 (/Users/scotch-83/.nvm/versions/node/v22.6.0/lib/node_modules/@ionic/cli)
   Ionic Framework               : @ionic/angular 8.3.0
   @angular-devkit/build-angular : 18.2.3
   @angular-devkit/schematics    : 18.2.3
   @angular/cli                  : 18.2.3
   @ionic/angular-toolkit        : 11.0.1

Capacitor:

   Capacitor CLI      : 6.1.2
   @capacitor/android : not installed
   @capacitor/core    : 6.1.2
   @capacitor/ios     : not installed

Utility:

   cordova-res : not installed globally
   native-run  : 2.0.1

System:

   NodeJS : v22.6.0 (/Users/scotch-83/.nvm/versions/node/v22.6.0/bin/node)
   npm    : 10.8.2
   OS     : macOS Unknown

### Additional Information

Ionic info did not find information about my mac:

MacOS Sonoma 14.5

Chip: Apple Silicon M2 Pro

### PR DESCRIPTION
**Pull request by tanner-reits**
Issue number: resolves ionic-team/stencil-ds-output-targets#476, resolves ionic-team/stencil-ds-output-targets#475, resolves #29848

---------

## What is the current behavior?

In v0.6.0 of the [React output target](https://www.npmjs.com/package/@stencil/react-output-target), the implementation was changed to leverage Lit's utility for creating React components from web components. This introduced some unforseen issues and breaking changes.

## What is the new behavior?

- Reverts many of the changes from https://github.com/ionic-team/ionic-framework/pull/29782 to downgrade the React output target package to the last stable version (v0.5.3)
- Downgrades the version of Stencil to v4.20.0 (due to https://github.com/ionic-team/stencil/issues/5983 causing problems with the downgraded output target)
- Pins these versions and prevents Renovate from attempting to upgrade until the associated issues are resolved

## Does this introduce a breaking change?

- [ ] Yes
- [X] No

## Other information

Dev build for this version: `8.3.1-dev.11726167750.15400355`

I tested the dev build against the use cases outlined in https://github.com/ionic-team/stencil-ds-output-targets/issues/475 and https://github.com/ionic-team/stencil-ds-output-targets/issues/476

### CODE REVIEW

File: renovate.json5 [hunk 1]
```diff
@@ -14,6 +14,20 @@
...
+      ],
+      allowedVersions: ["<=0.5.3"]
+    },
+    {
+      matchPackageNames: ["@stencil/core"],
+      matchFileNames: [
+        "core/package.json"
+      ],
+      allowedVersions: ["<=4.20.0"]
+    },
```
Comment (tanner-reits) [inline renovate.json5:30]: I _think_ this will work to prevent these packages from updating, but hard to know until it merges and we let Renovate run 🤷‍♂️

Comment (tanner-reits) [reply renovate.json5:30]: I'm also gonna create tasks for these respective issues so I'll add a comment with each of those task numbers once I have them

**Review by thetaPC — APPROVED** [review]
LGTM, verified that the React test build still works and that both bugs from stencil-ds-output-targets are resolved.

However, I couldn't replicate this [issue](https://github.com/ionic-team/ionic-framework/issues/29848). I highly suggest asking the community to see if the dev build fixes their issue. If it does then we can change the PR description to resolve the issue: `Issue number: ionic-team/stencil-ds-output-targets#476, ionic-team/stencil-ds-output-targets#475, #29848` to `Issue number: ionic-team/stencil-ds-output-targets#476, ionic-team/stencil-ds-output-targets#475, resolves #29848`

**Review by brandyscarney — APPROVED** [review]
Confirmed both of the linked issues in `stencil-ds-output-targets` are resolved by the dev build. 👍

**Pull request merged**